### DISCUSSION TITLE
Moving sigmoid opt to math has broken the import of pymc3

### DISCUSSION THREAD

**Discussion by danituckerpersonal** [aesara-devs/aesara#415] [Q&A]
Hello,

Because you recently moved the sigmoid opt to math, importing pymc3 breaks, as it looks for sigmoid in nnet still. Is this something you can fix? 

Thanks!

**Comment by OriolAbril** [comment]
I _think_ it is something that can be fixed, but if you are encountering this issue you are probably using the master branch from github for pymc, which for now you should avoid. The stable 3.x version is available at `v3` branch and the development of pymc3 4.0 is happening at `v4` branch, I'd recommend using one of these two branches to get pymc from github. 

`master` has some changes wrt to `v3` like using Aesara instead of Theano-pymc, but it is also missing some important bugfixes that are present in `v3`. `v4` is what really takes advantage of Aesara instead of treating it as a drop in replacement for Theano.

I know the advise is a bit tangential to the issue, but I think it's worth mentioning anyways.

**Reply by brandonwillard** [reply]
The current `master` branch of PyMC3 is definitely a major source for this problem, but we could also add a deprecation warning and an alias for the old `aesara.tensor.nnet.sigm` module, just in case other code was relying on it.

**Comment by danituckerpersonal** [comment]
Thank you all for your quick responses. I switched to v3 and got it working.

Thanks!

Dani

From: Brandon T. Willard ***@***.***>
Date: Monday, May 17, 2021 at 1:54 PM
To: pymc-devs/aesara ***@***.***>
Cc: Danielle Tucker ***@***.***>, Author ***@***.***>
Subject: Re: [pymc-devs/aesara] Moving sigmoid opt to math has broken the import of pymc3 (#415)

The current master branch of PyMC3 is definitely a major source for this problem, but we could also add a deprecation warning and an alias for the old aesara.tensor.nnet.sigm module, just in case other code was relying on it.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/pymc-devs/aesara/discussions/415#discussioncomment-749506>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/APUIPKAFMHYU3YVHYNMAJB3TOFQ4PANCNFSM45A34ZQA>.

Disclaimer

This is an email from iManage. The information contained in it and in any attachments is proprietary and confidential and is designated solely for the attention and use of the intended recipient(s). 
If you are not the intended recipient(s), please notify the sender immediately and then delete it (and any attachment) from your computer system(s). 
Any form of distribution, copying or use of this e-mail or any part of it is strictly prohibited. 
iManage does not accept legal responsibility for the contents of this e-mail and opinions expressed in it may not necessarily reflect those of the company. 
iManage does not accept liability for errors or omissions, or for any damage caused by viruses or other harmful programme routines.
